analysis/cli.py

```diff
@@ -207,47 +123,49 @@ def run_analysis():
         },
     }
-    download_ap = AnalysisProfile(
-        # in this case threshold_percent is the threshold for contribution to overall change
-        threshold_percent=1,
-        metric_name="downloads",
-        table_name="www_site_metrics_summary_v1",
-        dimensions=[
-            "country",
-        ],
-        sort_by=[
-            "contrib_to_overall_change",
-            "percent_change",
-            "change_to_contrib",
-            "percent_significance",
-        ],
-    )
 
-    analysis_profiles = [
-        (new_profiles_fenix_ap, new_profiles_fenix_date_ranges_of_interest),
-        (new_profiles_desktop_ap, new_profiles_desktop_date_ranges_of_interest),
-        (dau_fenix_ap, dau_fenix_date_ranges_of_interest),
-        (mau_firefox_desktop_ap, mau_firefox_desktop_date_ranges_of_interest),
-        (download_ap, download_date_ranges_of_interest),
-    ]
-
-    for (profile, date_ranges) in analysis_profiles:
-        significant_dims = find_significant_dimensions(profile=profile, date_ranges=date_ranges)
-        issue_report(
-            profile=profile,
-            evaluation=significant_dims,
-            date_ranges=date_ranges,
-        )
+    all_dates["Site Metrics Downloads"] = download_date_ranges_of_interest
+    return all_dates
+
 
+@cli.command()
+@click.argument("paths", required=True, type=click.Path(exists=True, file_okay=True), nargs=-1)
+def run_analysis(paths: Iterable[str]):
+    logger.info("Starting analysis")
+    date_ranges = _collect_time_ranges()
+    for path in paths:
+        configs = Loader.load_all_config_files(path)
+
+        # TODO GLE temp mapping of date range to Config
+        #  will be removed once data range config completed.
+        run_config = []
+        for config in configs:
+            run_config.append((config, date_ranges[config.analysis_profile.name]))
+
+        for (config, date_ranges) in run_config:
+            significant_dims = find_significant_dimensions(
+                profile=config.analysis_profile, date_ranges=date_ranges
+            )
+            issue_report(
+                profile=config.analysis_profile,
+                evaluation=significant_dims,
+                date_ranges=date_ranges,
+                notif_config=config.notification,
+            )
     logger.info("Analysis completed")
 
 
 @cli.command()
-@click.argument(
-    "config_files", required=True, type=click.Path(exists=True, file_okay=True), nargs=-1
-)
-def validate_config(config_files):
-    logger.info(f"Validating config files in: {config_files}")
+@click.argument("paths", required=True, type=click.Path(exists=True, file_okay=True), nargs=-1)
+def validate_config(paths: Iterable[str]):
+    """
+    Does not actually validate, only loads the config files.
+    """
+    logger.info(f"Validating config files in: {paths}")
+
+    for path in paths:
+        configs = Loader.load_all_config_files(path)
```

Comment:
If you'd like you could always keep track of any paths that result in some sort of Exception here and continue the looping. Then could log out list of all configs that failed to avoid having to fix and re-run the command just to see that the next path also fails.

Reply:
Yep - there's a lot of validation and handling of those validation errors that needs to be added.  Once the remaining changes to the config (the date ranges piece) is added I'll do a sweep of validation and error handling.